go18_test.go

```diff
@@ -155,6 +156,32 @@ func TestContextCancelQuery(t *testing.T) {
 	}
 }
 
+// TestIssue617 tests that a failed query in QueryContext doesn't lead to a
+// goroutine leak.
+func TestIssue617(t *testing.T) {
+	db := openTestConn(t)
+	defer db.Close()
+
+	const N = 10
+
+	numGoroutineStart := runtime.NumGoroutine()
+	for i := 0; i < N; i++ {
+		query := `SELECT * FROM DOESNOTEXIST`
+		ctx, _ := context.WithTimeout(context.Background(), time.Second)
+		rows, err := db.QueryContext(ctx, query)
+		if rows != nil || err == nil {
+			t.Fatal("expected nil rows and a non-nil error, got rows %v and error %v", rows, err)
+		}
+	}
+	numGoroutineFinish := runtime.NumGoroutine()
+
+	// We use N/2 and not N because the GC and other actors may increase or
```

Comment:
If you address my comment about leaking context goroutines, I think you'll be able to remove this (but please correct me if I'm wrong).